### PR TITLE
Chat link deletion button added.

### PR DESCRIPTION
**Pull request by manny525**

### CODE REVIEW

**Comment by Narasimha1997** [comment]
Can anyone delete the chatlink? If they have the chatlink`uuid` ??

**Comment by muke1908** [comment]
@Narasimha1997 you are correct but the chances of guessing right UUIDs are extremely small, it's not worth serious consideration. If you think carefully, chances of your password being rightly guessed are much higher.

Still, there are a few vulnerabilities in terms of DDoS, and it can be prevented by adding the rate limiter. 

I added "delete" feature keeping the following things in mind:
1. Chat link is meant for **only** receiver / sender.
2. Both receiver / sender will have the right to invalidate the chat link at any point in time.

Yes, if chat link is compromised there can be security issues but that is obvious, isn't it?
Even if I assume the chat link is compromised, then the only next best option for the user is to be able to invalidate the link so that other people can't impersonate. 

In terms of vulnerability, the attacker can only delete a link.

**Comment by Narasimha1997** [comment]
@muke1908 , Yes you are right. Of coure UUID can't be guessed, I didn't mean it can be guessed. 
 The delete api takes chatlink UUID directly to delete the record. The one who possess chatlink can do anything according to the current architecture, they can join the chat as well as delete it. 

The API returns Chatlink to the frontend according to issue #159. One suggestion would be to avoid sending chatlink to frontend or avoid sending chatlink in clear text (you can encrypt it with the PIN)

One more suggestion: Can we keep PIN validation for delete api? So that it would be like a double check, if someone steals the chatlink, they still need PIN to delete the chatlink.

**Comment by muke1908** [comment]
>The API returns Chatlink to the frontend according to issue #159. One suggestion would be to avoid sending chatlink to frontend or avoid sending chatlink in clear text (or you can just send the hash of chatlink), in other words avoid frontend from knowing the actual chatlink.

But anyway we'll have to redirect them to the chat link frontend needs to know the link.

>One more suggestion: Can we keep PIN validation for delete api? So that it would be like a double check, if someone steals the chatlink, they still need PIN to delete the chatlink.

Yes, this can add an extra layer of security but, the user will have to memorize the PIN. But I somewhat feel, unnecessary.

**Comment by manny525** [comment]
Do let me know what changes are required and I'll try to get them done

**Comment by muke1908** [comment]
> Do let me know what changes are required and I'll try to get them done

Hi, I didn't get time to review the PR completely. As I understood, you are showing the delete button only when the link is active, but the user should be able to delete the link at any point of time. 

Also, please resolve the merge conflicts.

File: .env.sample [hunk 1]
```diff
@@ -1,3 +1,4 @@
+GOOGLE_RECAPTCHA_SECRET=''
```
Comment (muke1908) [inline .env.sample:1]:
```suggestion
```

File: client/src/components/Messaging/LinkSharingInstruction/index.js [hunk 2]
```diff
@@ -1,16 +1,32 @@
...
+const LinkSharingInstruction = ({ online, link, handleDeleteLink }) => {
   const inputRef = useRef(null);
+  const [darkMode] = useContext(ThemeContext);
 
   const clickHandler = () => {
     inputRef.current.select();
   };
+  const deleteHandler = async () => {
+    if (window.confirm('Delete chat link forever?')) await handleDeleteLink();
+  };
```
Comment (muke1908) [inline client/src/components/Messaging/LinkSharingInstruction/index.js:15]: Let's show some loading state, then redirect to homepage.

File: client/src/components/Messaging/LinkSharingInstruction/index.js [hunk 3]
```diff
@@ -1,16 +1,32 @@
...
+      <div className={deleteLinkButton}>
+        <div
+          className={`${styles.sendButton} ${!darkMode && styles.lightModeSend}`}
+          type="submit"
+          role="button"
+          onClick={deleteHandler}
+        >
+          Delete
+        </div>
+      </div>
```
Comment (muke1908) [inline client/src/components/Messaging/LinkSharingInstruction/index.js:29]: Please move it outside. 'Delete link' is not really related with `LinkSharingInstruction` component. Also, we need to display the delete button always. Let's add the button to the top right corner - next to theme toggle button.

File: client/src/service/deleteLink.js [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import makeRequest from '../utils/makeRequest';
+
+const getLink = async ({ channelID }) => {
```
Comment (muke1908) [inline client/src/service/deleteLink.js:3]: `deleteLink`

**Review by muke1908 — CHANGES_REQUESTED** [review]

**Comment by muke1908** [comment]
@manny525 Hi, I have left a few comments. Please check.

**Comment by manny525** [comment]
On it @muke1908

**Comment by muke1908** [comment]
Hi @manny525, the build is failing in your last commit. Can you please fix this?

**Pull request merged**